### PR TITLE
Separate Alluxio related documentation

### PR DESCRIPTION
**Pull request by mosabua**
Since we are adding caching docs as a separate document . the Alluxio docs should sit side by side with it and not be in the main Hive connector docs. We plan to pull that long document apart even more in the future but this is necessary now since we have built in caching support. 

Related to https://github.com/prestosql/presto/pull/4227

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/hive-alluxio-caching.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
+===========================================
+Hive Connector Storage Caching with Alluxio
```
[the remark this answers is not in the document]
Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive-alluxio-caching.rst]: Sure .. but should we rename the file as well then? Also do we have any docs about the catalog services?

File: presto-docs/src/main/sphinx/connector/hive-alluxio-caching.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+Presto queries will then transparently retrieve and cache files or objects from
+a variety of disparate storage systems including HDFS and S3.
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Alluxio client-side configuration
```
[the remark this answers is not in the document]
Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive-alluxio-caching.rst]: Leaving lower case to follow Google style

File: presto-docs/src/main/sphinx/connector/hive-alluxio-caching.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+leveraging Alluxio's distributed block-level read/write caching functionality.
+The tables must be created in the Hive metastore with the ``alluxio://``
+location prefix (see `Running Apache Hive with Alluxio
+<https://docs.alluxio.io/os/user/2.1/en/compute/Hive.html?utm_source=prestosql&utm_medium=prestodocs>`_
+for details and examples).
+
+Presto queries will then transparently retrieve and cache files or objects from
+a variety of disparate storage systems including HDFS and S3.
+
+.. contents::
```
[the remark this answers is not in the document]
Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive-alluxio-caching.rst]: Done

File: presto-docs/src/main/sphinx/connector/hive-alluxio-caching.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive-alluxio-caching.rst]: Done already actually.

**Comment by electrum** [comment]
Can you rebase?

**Comment by mosabua** [comment]
Rebased .. somehow that pulled in the contributing guideline changes.. should I reissue PR from a new branch?

**Comment by electrum** [comment]
I think GitHub is just confused.

**Pull request merged**